src/DotNetLightning.Core/Utils/Primitives.fs

```diff
@@ -567,20 +596,37 @@ module Primitives =
                 raise <| FormatException(sprintf "Failed to parse %s" s)
             )
 
+    [<Obsolete>]
     type UserId = UserId of uint64
+
+    [<Obsolete>]
     type Delimiter = Delimiter of array<byte>
 
+    /// Each lightning node can specify its own RGB color in node_announcement.
+    /// Used for e.g. lightning node explorer.
+    /// <seealso cref="DotNetLightning.Serialization.Msgs.UnsignedNodeAnnouncementMsg" />
+    [<Struct>]
```

Comment:
this Struct addition is unrelated?

Reply:
I have split into independent commit.
(Please don't say we should create another PR. independent commit is enough.)